src/test/SecurityPanel/StoreJmxCredentials.test.tsx

```diff
@@ -0,0 +1,256 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import userEvent from '@testing-library/user-event';
+import renderer, { act } from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import { of } from 'rxjs';
+import { Target } from '@app/Shared/Services/Target.service';
+
+import '@testing-library/jest-dom';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+import { NotificationMessage } from '@app/Shared/Services/NotificationChannel.service';
+
+const mockTarget: Target = { connectUrl: 'service:jmx:rmi://someUrl', alias: 'fooTarget' };
+const mockAnotherTarget: Target = { connectUrl: 'service:jmx:rmi://anotherUrl', alias: 'anotherTarget' };
+
+jest.mock('@app/SecurityPanel/CreateJmxCredentialModal', () => {
+  return {
+    CreateJmxCredentialModal: jest.fn((props) => {
+      return (
+        <Modal
+          isOpen={props.visible}
+          variant={ModalVariant.large}
+          showClose={true}
+          onClose={props.onClose}
+          title="CreateJmxCredentialModal"
+        >
+          Jmx Auth Form
+        </Modal>
+      );
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/NotificationChannel.service', () => {
+  const mockNotification = { message: { target: 'service:jmx:rmi://someUrl' } } as NotificationMessage;
+  return {
+    ...jest.requireActual('@app/Shared/Services/NotificationChannel.service'),
+    NotificationChannel: jest.fn(() => {
+      return {
+        messages: jest
+          .fn()
+          .mockReturnValueOnce(of()) // 'renders correctly'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'adds the correct table entry when a stored notification is received'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of(mockNotification))
+
+          .mockReturnValueOnce(of()) // 'removes the correct table entry when a deletion notification is received'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'renders an empty table after receiving deletion notifications for all credentials'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValue(of()), // all other tests
+      };
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/Target.service', () => {
+  return {
+    ...jest.requireActual('@app/Shared/Services/Target.service'),
+    TargetService: jest.fn(() => {
+      return {
+        deleteCredentials: jest.fn(),
+      };
+    }),
+  };
+});
+
+jest.doMock('@app/Shared/Services/Api.service', () => {
```

Comment:
Just realized that this can be changed back to `jest.mock`. Even though it'll be hoisted above the `const mockTarget` and `const mockAnotherTarget` variable definitions, the actual `jest.fn` implementations inside the mocks aren't invoked until we do the imports on line 145/146, and by that time the variables would've been defined.

Reply:
Oh nice, I'll change it back